### ISSUE TITLE
Remove `std::function<>` from the `Ledger::GetExternalWallet()` flow

### ISSUE THREAD

**Issue by szilardszaloki** [brave/brave-browser#24108]
Part2 of https://github.com/brave/brave-browser/issues/18560.

**Comment by GeetaSarvadnya** [comment]
Verification is inprogress

```

Brave | 1.43.63 Chromium: 104.0.5112.81 (Official Build) beta (64-bit)
-- | --
Revision | 5b7b76419d50f583022568b6764b630f6ddc9208-refs/branch-heads/5112@{#1309}
OS | Windows 10 Version 21H2 (Build 19044.1889)

```

**Comment by GeetaSarvadnya** [comment]
Unable to login Gemini accounts and claim ad grants on bitFlyer is throwing error. Adding QA block until these issues resolved

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.43.79 Chromium: 104.0.5112.102 (Official Build) beta (x86_64)
-- | --
Revision | 8e5396254975ef939f2ef7d0bd334e48a052b536-refs/branch-heads/5112@{#1478}
OS | macOS Version 12.5.1 (Build 21G83)
```

**Comment by Uni-verse** [comment]
Verification in progress on Samsung Galaxy S21 using the following build(s):

Brave	1.43.84 Chromium: 105.0.5195.68 (Official Build) (64-bit) 
Revision	ad13e82529051bac6a0e65f455e6d7a1e5fd7938-refs/branch-heads/5195@{#903}
OS	Android 12; Build/SP1A.210812.016

<details><summary>Gemini</summary>

- Confirmed Gemini linking with brave wallet works as expected
- Confirmed userdrain works as expected
- Confirmed that claimed vBAT is moved to Gemini wallet and balance is shown in the wallet (note, waiting period with Gemini)
- Confirmed balance is updated as expected in UI (brave://rewards, panel, brave://rewards-internals)

</details>

<details><summary>Uphold</summary>

- Confirmed Uphold linking with brave wallet works as expected
- Confirmed userdrain works as expected
- Confirmed that claimed vBAT is moved to Gemini wallet and balance is shown in the wallet (note, waiting period with Gemini)
- Confirmed balance is updated as expected in UI (brave://rewards, panel, brave://rewards-internals)

</details>

<details><summary>bitFlyer</summary>

- Confirmed bitFlyer linking with brave wallet works as expected
- Confirmed userdrain works as expected
- Confirmed that claimed vBAT is moved to Gemini wallet and balance is shown in the wallet (note, waiting period with Gemini)
- Confirmed balance is updated as expected in UI (brave://rewards, panel, brave://rewards-internals)

1 | 2 | 3 | 4 | 5 | 6
-|-|-|-|-|-
![screenshot-1661892686088](https://user-images.githubusercontent.com/17885425/187542528-c1cad2c3-73fc-4d9b-9d29-2382e342f128.png) | ![screenshot-1661893064334](https://user-images.githubusercontent.com/17885425/187542533-b7bb032f-df5a-448d-8c6d-58edf38fd20a.png) | ![screenshot-1661893087558](https://user-images.githubusercontent.com/17885425/187543224-06a50e5c-5b82-4d1e-94cf-7547cec24bcf.png) | ![screenshot-1661893097868](https://user-images.githubusercontent.com/17885425/187543228-87382e40-469a-4d15-8c58-d3ef356910b8.png) | ![screenshot-1661893109212](https://user-images.githubusercontent.com/17885425/187542546-3773a472-d145-4c56-9981-495443705620.png) | ![screenshot-1661893314673](https://user-images.githubusercontent.com/17885425/187542548-f7449986-f0a8-4354-a73c-16667ab68a02.png)


</details>

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.43.88 Chromium: 105.0.5195.68 (Official Build) (64-bit)
-- | --
Revision | ad13e82529051bac6a0e65f455e6d7a1e5fd7938-refs/branch-heads/5195@{#903}
OS | Ubuntu 18.04 LTS


- Verified the test plan from https://github.com/brave/brave-core/pull/14218


<details>
<summary>Gemini - PASSED</summary>

- Confirmed `Gemini linking` with brave wallet works as expected
- Confirmed `userdrain` works as expected
- Confirmed that claimed vBAT is moved to Gemini wallet and balance is shown in the wallet (note, waiting period with Gemini)
- Confirmed balance is updated as expected in UI (brave://rewards, panel, NTP, brave://rewards-internals)

Ad Grant | Ad Grant | Balance after claiming 
----------|----------|-----------
![image](https://user-images.githubusercontent.com/34715963/187911874-b468e295-01da-4f88-8e0c-9ff39d60d5b4.png)|![image](https://user-images.githubusercontent.com/34715963/187911914-8c572388-32a9-430a-a093-c3d3f3ea26a6.png)|![image](https://user-images.githubusercontent.com/34715963/187912025-145ab8c8-9af7-4a9c-9049-64fe4ba4c2e4.png)


After linking
![image](https://user-images.githubusercontent.com/34715963/187912226-bd1f9123-4bf2-467f-bb2a-fc21b93db0cc.png)
![image](https://user-images.githubusercontent.com/34715963/187912279-e5cfde57-83ce-4a83-b931-71df9e8b3816.png)



_vBAT drain event logs:_
![image](https://user-images.githubusercontent.com/34715963/187912344-1e6f5919-89de-441f-8c82-3e943ba01f37.png)



_Console logs for drain id after userdrain was successful:_

```
[ REQUEST ]
> URL: https://grant.rewards.bravesoftware.com/v2/suggestions/claim
> Method: POST
> Content: {"credentials":[{"publicKey":"WNbOpSKV68gOBVnMEqt0Oqnb+DYTn


[ RESPONSE - OnRequest ]
> Url: https://grant.rewards.bravesoftware.com/v2/suggestions/claim
> Result: Success
> HTTP Code: 200
> Body: {"drainId":"9b7ea823-27d5-4960-a5fb-73b6da154b6b"}

```
_Gemini final balance after the vBAT user drain_

```
[ RESPONSE - OnRequest ]
> Url: https://api.sandbox.gemini.com/v1/balances
> Result: Success
> HTTP Code: 200
> Body: [{"type":"exchange","currency":"BAT","amount":"1502","available":"1502","availableForWithdrawal":"1502"},

```

</details>

<details>
<summary>Uphold - PASSED</summary>

- Confirmed `Uphold linking` with brave wallet works as expected
- Confirmed `userdrain` works as expected
- Confirmed that claimed vBAT is moved to Uphold wallet and balance is shown in the wallet 
- Confirmed balance is updated as expected in UI (brave://rewards, panel, NTP, brave://rewards-internals)

Ad Grant | Ad Grant | Balance after claiming 
----------|----------|-----------
![image](https://user-images.githubusercontent.com/34715963/187913597-2c9964ce-4be5-421f-9bc7-b469b7783232.png)|![image](https://user-images.githubusercontent.com/34715963/187913643-9ab46353-5fba-4278-93fb-8efd627dbfcc.png)|![image](https://user-images.githubusercontent.com/34715963/187913733-d49cb628-de35-4225-aa31-774e28c39644.png)


after linking
![image](https://user-images.githubusercontent.com/34715963/187914331-cfa65068-648d-4b5f-a2af-db2c29524ad9.png)
![image](https://user-images.githubusercontent.com/34715963/187914405-23e94444-55ad-4f7f-a817-b456bdf4a65f.png)


_vBAT drain event logs:_

![image](https://user-images.githubusercontent.com/34715963/187914465-a209fc51-076a-4cd4-bc7c-8083e31a1c40.png)

_Console logs for drain id after userdrain was successful:_

```
[ RESPONSE - OnRequest ]
> Url: https://grant.rewards.bravesoftware.com/v2/suggestions/claim
> Result: Success
> HTTP Code: 200
> Body: {"drainId":"79b96335-b9c9-4982-af97-0ab6d6abb0c3"}

```

_Uphold final balance after the vBAT user drain_

```
[ RESPONSE - OnRequest ]
> Url: https://api-sandbox.uphold.com/v0/me/cards/c8efcc83-749d-4fa8-a5e8-6be943518eb4
> Result: Success
> HTTP Code: 200
> Body: {"CreatedByApplicationId":"193a77cf-02e8-4e10-8127-8a1b5a8bfece","address":{"wire":"UH38EFC9B9"},"available":"2.00","balance":"2.00","currency":"BAT","id":"c8efcc83-749d-4fa8-a5e8-6be943518eb4","label":"Brave Browser","
```

</details>

<details>
<summary>bitFlyer - PASSED</summary>

- Confirmed `bitFlyer linking` with brave wallet works as expected
- Confirmed `userdrain` works as expected
- Confirmed that claimed vBAT is moved to bitFlyer wallet and balance is shown in the wallet 
- Confirmed balance is updated as expected in UI (brave://rewards, panel, NTP, brave://rewards-internals)

Ad Grant | Ad Grant | Balance after claiming 
----------|----------|-----------
![image](https://user-images.githubusercontent.com/34715963/187916174-d037e97c-1467-4971-9dbb-c00918053be0.png)|![image](https://user-images.githubusercontent.com/34715963/187916495-61712ec3-d775-4322-be61-652f0ca7babe.png)|![image](https://user-images.githubusercontent.com/34715963/187916566-5bf814e1-3bf6-4ece-898f-cddf77d59616.png)


after linking
![image](https://user-images.githubusercontent.com/34715963/187920454-418087ea-3eda-4d99-be50-293c87b20401.png)
![image](https://user-images.githubusercontent.com/34715963/187920634-7bd10c6a-4e20-4470-bcf5-07554b90e071.png)



_vBAT drain event logs:_
![image](https://user-images.githubusercontent.com/34715963/187920675-b75ca0bb-39a3-4c45-8c45-1666f749ab59.png)


_Console logs for drain id after userdrain was successful:_

```
[ RESPONSE - OnRequest ]
> Url: https://grant.rewards.bravesoftware.com/v2/suggestions/claim
> Result: Success
> HTTP Code: 200
> Body: {"drainId":"1baecf77-cf29-4aa6-b438-ff10c4963548"}

```

_bitFlyer final balance after the vBAT user drain_

```
[ RESPONSE - OnRequest ]
> Url: https://demo24kiuw4dcyncsy3qlud8u8.azurewebsites.net/api/link/v1/account/inventory
> Result: Success
> HTTP Code: 200
> Body: {
  "account_hash": "0c2d0b37e33da34cc82d8659ff8110d8095b9bd3",
  "inventory": [
    {
      "currency_code": "BAT",
      "amount": 102.0,
      "available": 102.000000000000
    },


```

</details>
